### DISCUSSION TITLE
Stripe Webhook triggering multiple times

### DISCUSSION THREAD

**Discussion by EsotericCoder** [pay-rails/pay#674] [Q&A]
I configured  Pay::Webhooks.delegator.subscribe "stripe.checkout.session.completed", BookingCompleted.new 

The issue is this is triggering at least four times. What is the best way to only account for it once.

class BookingCompleted
  def call(event)
    object = event.data.object

    if object.payment_status == "paid"
      booking = Booking.find_by_stripe_session_id(object.id)
      booking.paid!

      # Store Charge ID
      payment_intent = Stripe::PaymentIntent.retrieve(object.payment_intent)
      charge = payment_intent.charges.data.first
      booking.update(stripe_charge_id: charge.id, stripe_receipt_url: charge.receipt_url)

      # Queue Payout
      PayOutJob.set(wait_until: booking.start_time + 1.hour ).perform_later(booking)
    end
  end
end

**Comment by excid3** [comment]
Check the status of your booking and skip it if it's completed.
